atc/db/resource_type.go

```diff
@@ -87,11 +90,24 @@ func (resourceTypes ResourceTypes) Deserialize() atc.VersionedResourceTypes {
 	var versionedResourceTypes atc.VersionedResourceTypes
 
 	for _, t := range resourceTypes {
+		// Apply source defaults to resource types
+		source := t.Source()
+		parentType, found := resourceTypes.Parent(t)
+		if found {
+			source = parentType.Defaults().Merge(source)
+		} else {
+			parentType, found, _ := t.FindParentBaseResourceType()
+			if found {
+				source = parentType.Defaults.Merge(source)
```

Comment:
Shadowing `parentType` confused me as to why it was `parentType.Defaults` in one place and `parentType.Defaults()` in another
		
```suggestion
			parentUbrt, found, _ := t.FindParentBaseResourceType()
			if found {
				source = parentUbrt.Defaults.Merge(source)
```

Reply:
With the second commit, this is no longer an issue.